orttraining/orttraining/training_api/onnxruntime_training_c_api.cc

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/framework/error_code_helper.h"
+#include "core/framework/ort_value.h"
+#include "core/session/ort_apis.h"
+#include "core/session/ort_env.h"
+#include "orttraining/training_api/include/checkpoint.h"
+#include "orttraining/training_api/include/training_session.h"
+#include "core/session/abi_session_options_impl.h"
+
+ORT_API_STATUS_IMPL(OrtApis::CreateTrainingSession, _In_ const OrtEnv* env, _In_ const OrtSessionOptions* options,
+                    _Inout_ OrtCheckpointState* checkpoint_state, _Outptr_ OrtTrainingSession** out) {
+  API_IMPL_BEGIN
+  std::unique_ptr<onnxruntime::training::api::TrainingSession> train_sess;
+  auto chkpt_state = reinterpret_cast<onnxruntime::training::api::CheckpointState*>(checkpoint_state);
+  OrtStatus* status = nullptr;
+  *out = nullptr;
+
+  ORT_TRY {
+      train_sess = std::make_unique<onnxruntime::training::api::TrainingSession>(
+        env->GetEnvironment(),
```

Comment:
Shall we check for env being a `nullptr`?

Reply:
why specifically this one? All others are pointers too

Reply:
I thought we had a check for all. I saw a check for session options, and thought that we should add a check for env also.